File: CSL-101 Computer Programming/CP_Assignment_5/ABD_5.py
# Assignment-5
# Q)5


def fact(n):
    return 1 if (n == 1 or n == 0) else n * fact(n - 1)


n = int(input("Enter the number : "))
m = n
n1 = 0
while m > 0:
    r = m % 10
    n1 += fact(r)
    m //= 10
print("It is a Strong no.") if n == n1 else print("It is not a Strong no.")

File: CSL-101 Computer Programming/CP_Assignment_4/ABD_5.c
// Assignment-4
// Q)5

# include <stdio.h>
# include <ctype.h>

int main()
{
    int upper, lower, number, special;
    upper = lower = number = special = 0;
    char filename[128], ch;
    printf("Enter the file name to open: ");
    scanf(" %[^\n]s", filename);
    FILE *file = fopen(filename, "r");
    if (file == NULL)
    {
        printf("Error opening the File");
        return 1;
    }
    while ((ch = fgetc(file)) != EOF)
    {
        if (isupper(ch))
        {
            upper++;
        }
        else if (islower(ch))
        {
            lower++;
        }
        else if (isdigit(ch))
        {
            number++;
        }
        else
        {
            special++;
        }
    }
    printf("There are %d UpperCase Letters\n"
           "There are %d LowerCase Letters\n"
           "There are %d Numbers\n"
           "There are %d Special characters\n",
           upper, lower, number, special);
    fclose(file);
    return 0;
}